lib/metasploit/framework/login_scanner/jenkins.rb

```diff
@@ -5,50 +5,133 @@ module Framework
     module LoginScanner
       # Jenkins login scanner
       class Jenkins < HTTP
-
-        include Msf::Exploit::Remote::HTTP::Jenkins
-
         # Inherit LIKELY_PORTS,LIKELY_SERVICE_NAMES, and REALM_KEY from HTTP
         CAN_GET_SESSION = true
-        DEFAULT_PORT    = 8080
-        PRIVATE_TYPES   = [ :password ]
+        DEFAULT_HTTP_NOT_AUTHED_CODE = 403
+        DEFAULT_PORT = 8080
+        PRIVATE_TYPES = [:password].freeze
+        JENKINS_LOGIN_VALIDATION_URL = 'j_spring_security_check'.freeze
+
+        # Checks the setup for the Jenkins Login scanner.
+        # Always returns false, as the underlying client does not support checking the
+        # conditions due Jenkins not doing Authentication challenges.
+        # This is needed for Pro.
+        #
+        # @return [Boolean] Always returns false.
+        def check_setup
+          false
+        end
```

Comment:
can we do a better check than this? I get that we need to override the default in the http login scanner but surely there are some checks we can make to verify that the service is up for example

Reply:
this is only called from Pro, and the original code didn't even check any parameters passed from pro, so not this does not need be better